arcgis-ios-sdk-samples/Analysis/Viewshed (GeoElement)/ViewshedGeoElementViewController.swift

```diff
@@ -41,42 +52,57 @@ class ViewshedGeoElementViewController: UIViewController, AGSGeoViewTouchDelegat
         let elevationSource = AGSArcGISTiledElevationSource(url: brestElevationServiceURL)
         surface.elevationSources.append(elevationSource)
         scene.baseSurface = surface
-
+        
         /// The url of the scene service for buildings in Brest, France.
         let brestBuildingsServiceURL = URL(string: "https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/Buildings_Brest/SceneServer/layers/0")!
         // add a scene layer
         let buildings = AGSArcGISSceneLayer(url: brestBuildingsServiceURL)
         scene.operationalLayers.add(buildings)
-
+        
+        return scene
+    }
+    
+    private func addGraphicsOverlay() {
```

Comment:
Could you please make these functions factory methods like you did with `makeScene()`?

`addGraphicsOverlay()` -> `makeGraphicsOverlay()`
`addCameraController()` -> `makeCameraController()`

That way the configuration of the scene view (setting the scene, operational layers, and camera controller) all happens in one place.

Reply:
Good call!